src/client/partition.rs

```diff
@@ -32,8 +32,40 @@ use tracing::{error, info};
 
 use super::error::ServerErrorResponse;
 
+/// How strongly a [`PartitionClient`] is bound to a partition.
+///
+/// Under some circumstances and broker implementations, you might face a [`ProtocolError::UnknownTopicOrPartition`]
+/// for nearly all operations directly after topic/partition creation. Since the Kafka protocol offers no (at least
+/// to our knowledge) we to differentiate between "a topic/partition actually does not exist" and "our multiverse
+/// split brain is not synced up yet", we can only offer your this manual way to wait for the existence.
+///
+/// We offer the following ways to deal with this:
+///
+/// - Use a [strong binding](Self::Strong) which assumes a partition exists and retries [`ProtocolError::UnknownTopicOrPartition`]
+/// - Use a [weak binding](Self::Weak). All other methods (including the creation of a [`PartitionClient`]) may produce
+///   sporadic [`ProtocolError::UnknownTopicOrPartition`] errors.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum PartitionClientBindMode {
+    /// Assume that the partition (or topic that contains it) can be deleted at any time.
+    ///
+    /// This bubbles [`ProtocolError::UnknownTopicOrPartition`] to the user.
+    Weak,
+
+    /// Assume that the partition (and topic that contais it) exist.
+    ///
+    /// This retries [`ProtocolError::UnknownTopicOrPartition`].
+    Strong,
+}
```

Comment:
I find the `bind_mode` / `PartitionClientBindMode::Strong` (and weak) naming to be quite disconnected from the underlying behaviour they control - what about something like:

```suggestion
pub enum UnknownTopicHandling {
    /// When a [`ProtocolError::UnknownTopicOrPartition`] is returned by Kafka,
    /// it is passed up to the user to handle.
    ///
    /// These errors may occur spuriously.
    Error,

    /// Always retry operations that return a
    /// [`ProtocolError::UnknownTopicOrPartition`], until the operation succeeds
    /// or a different error is returned.
    ///
    /// This may unpredictably increase operation latency.
    Retry,
}
```

Reply:
done